huma_signals/adapters/ethereum_wallet/adapter.py

```diff
@@ -1,94 +1,88 @@
-    required_inputs: ClassVar[List[str]] = ["borrower_wallet_address"]
-    signals: ClassVar[List[str]] = list(EthereumWalletSignals.__fields__.keys())
+    required_inputs: ClassVar[list[str]] = ["borrower_wallet_address"]
+    signals: ClassVar[list[str]] = list(EthereumWalletSignals.__fields__.keys())
 
     def __init__(
         self,
+        eth_client_: eth_client.BaseEthClient | None = None,
         etherscan_base_url: str = settings.etherscan_base_url,
         etherscan_api_key: str = settings.etherscan_api_key,
     ) -> None:
-        if not etherscan_base_url:
-            raise ValueError("etherscan_base_url is empty")
-
-        self.etherscan_base_url = etherscan_base_url
-        self.etherscan_api_key = etherscan_api_key
+        self.eth_client = eth_client_ or eth_client.EthClient(
+            etherscan_base_url=etherscan_base_url,
+            etherscan_api_key=etherscan_api_key,
+        )
 
     async def fetch(  # pylint: disable=arguments-differ
         self, borrower_wallet_address: str, *args: Any, **kwargs: Any
     ) -> EthereumWalletSignals:
-        raw_txns = await self._node_get_transactions(
+        transactions = await self.eth_client.get_transactions(
             borrower_wallet_address,
         )
-        txn_df = pd.DataFrame.from_records(raw_txns)
-        if len(txn_df) == 0:
-            return EthereumWalletSignals(
-                total_transactions=0,
-                total_sent=0,
-                total_received=0,
-                wallet_tenure_in_days=0,
-                total_income_90days=0,
-                total_transactions_90days=0,
-            )
-        now = datetime.datetime.utcnow()
-        txn_df["timeStamp"] = pd.to_datetime(txn_df["timeStamp"], unit="s")
-        txn_df["value"] = txn_df["value"].astype(float)
-        txn_df["from"] = txn_df["from"].str.lower()
-        txn_df["to"] = txn_df["to"].str.lower()
-        txn_df["is_sent"] = txn_df["from"] == borrower_wallet_address.lower()
-        txn_df["is_received"] = txn_df["to"] == borrower_wallet_address.lower()
-        txn_df["in_90days"] = txn_df["timeStamp"] > now - pd.Timedelta(days=90)
-        # TODO: Limit to selected set of tokens
-        txn_df["income"] = txn_df["value"] * txn_df["is_received"].astype(float)
+        now = datetime_utils.tz_aware_utc_now()
+        min_tx_timestamp = (
+            None
+            if len(transactions) == 0
+            else min(tx.time_stamp for tx in transactions)
+        )
 
         return EthereumWalletSignals(
-            total_transactions=len(txn_df),
-            total_sent=sum(txn_df["is_sent"]),
-            total_received=sum(txn_df["is_received"]),
-            wallet_tenure_in_days=(now - txn_df["timeStamp"].min()).days,
-            total_income_90days=sum(txn_df["income"] * txn_df["in_90days"]),
-            total_transactions_90days=sum(txn_df["in_90days"]),
+            total_transactions=len(transactions),
+            total_sent=len(
```

Comment:
@PlayJok3r so the alternative to pandas is doing list comprehensions. Lemme know if your thoughts on this?